projects/onto-search/src/lib/onto-search-facet/dropdown-multi-select-facet/dropdown-multi-select/dropdown-multi-select.component.ts

```diff
@@ -0,0 +1,177 @@
+
+  private previousSelected: any[];
+  private change = new EventEmitter<string>();
+
+  constructor(@Inject(MatSelect) public matSelect: MatSelect,
+              private changeDetectorRef: ChangeDetectorRef) {
+    super();
+  }
+
+  ngOnInit(): void {
+    const panelClass = 'mat-select-search-panel';
+    this.matSelect.panelClass = panelClass;
+
+    this.matSelect.openedChange
+        .pipe(untilComponentDestroyed(this))
+        .subscribe((opened) => opened ? this.focus() : this.reset());
+
+    // detect changes when the input changes
+    this.change
+        .pipe(untilComponentDestroyed(this))
+        .subscribe(() => this.changeDetectorRef.detectChanges());
+
+    this.init();
+  }
+
+  /**
+   * Handles the key down event with MatSelect.
+   * Do not handles SPACE or ENTER key press
+   * @param {KeyboardEvent} event
+   */
+  handleKeydown(event: KeyboardEvent): void {
+    if (event.keyCode === ENTER || event.keyCode === SPACE) {
+      event.stopPropagation();
+    }
+  }
+
+  writeValue(value: string): void {
+    const valueChanged = value !== this.value;
+    if (valueChanged) {
+      this.value = value;
+      this.change.emit(value);
+    }
+  }
+
+  onInputChange(value): void {
+    const valueChanged = value !== this.value;
+    if (valueChanged) {
+      this.value = value;
+      this.onChange(value);
+      this.change.emit(value);
+    }
+  }
+
+  onBlur(value: string): void {
+    this.writeValue(value);
+    this.onTouched();
+  }
+
+  registerOnChange(fn: Function): void {
+    this.onChange = fn;
+  }
+
+  registerOnTouched(fn: Function): void {
+    this.onTouched = fn;
+  }
+
+  /**
+   * Focuses the search input field
+   * @private
+   */
+  public focus(): void {
+    if (!this.searchSelectInput) {
+      return;
+    }
+    this.searchSelectInput.nativeElement.focus();
+  }
+
+  /**
+   * Resets the current search value
+   * @param {boolean} focus whether to focus after resetting
+   * @private
+   */
+  public reset(focus?: boolean): void {
+    if (!this.searchSelectInput) {
+      return;
+    }
+    this.searchSelectInput.nativeElement.value = '';
+    this.onInputChange('');
+    if (focus) {
+      this.focus();
+    }
+  }
+
+  /**
+   * Initializes handling <mat-select [multiple]="true">
+   * Note: to improve this code, mat-select should be extended to allow disabling resetting the selection while filtering.
+   */
+  private init(): void {
+    // if <mat-select [multiple]="true">
+    // store previously selected values and restore them when they are deselected
+    // because the option is not available while we are currently filtering
+    this.matSelect.valueChange
+        .pipe(untilComponentDestroyed(this))
+        .subscribe((values) => {
+          if (this.matSelect.multiple) {
+            let restoreSelectedValues = false;
+            if (this.value && this.value.length && this.previousSelected && Array.isArray(this.previousSelected)) {
```

Comment:
`this.previousSelected` can be omited because `Array.isArray` will return false in case argument is undefined